CRM/Contribute/BAO/Contribution.php

```diff
@@ -3804,6 +3804,15 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
       $participantId = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_ParticipantPayment', $contributionId, 'participant_id', 'contribution_id');
     }
 
+    // fetch membership IDs registered against given contribution ID
```

Comment:
So, I feel like the changes in this file, along with their test, should stand alone as a discrete PR & be fairly easy to get merged. My worry is that this seems to duplicate the code called by completeOrder, but do a bit less. Extract & share?

Reply:
yup will extract this piece of code along with related UT on a separate PR